Comment out unecessary discriminatoras

jadams-tresys opened the pull request:
Only non-whitespace changes are on line 451, 525, and 542.

These discriminator statements might be able to be placed inside a subsequence to get around the changes from DAFFODIL-1971, but seem unnecessary in general.

stevedlawrence reviewed (commented):
Can you revert unrelated white space changes? If we do want to clean up the white space, it should be done in a separate commit.

mbeckerle commented on src/main/resources/EDIFACT-Common/EDIFACT-Service-Segments-4.1.xsd:
```diff
@@ -441,25 +441,25 @@
   <xsd:complexType name="UIH-InteractiveMessageHeader">
     <xsd:sequence dfdl:ref="ibmEdiFmt:EDISegmentSequenceFormat">
-      <xsd:annotation><xsd:appinfo source="http://www.ogf.org/dfdl/">
+      <!--<xsd:annotation><xsd:appinfo source="http://www.ogf.org/dfdl/">
```
Add a comment as to why these are unnecessary.

mbeckerle reviewed (approved):
+1
I just think add comment as to why these discrims are no longer needed.

Commit:
Use nested sequences for discriminabors

With the changes in Daffodil for DAFFODIL-1971 we need to use nested
sequences to maintain the same discriminator as previously. In this
case, the discriminators were simply using fn:true(), which prior to
DAFFODIL-1971 would only be executed if the rest of the body
successfully parsed. After DAFFODIL-1971 all discriminator expressions
are attempted regardless if the rest of the sequence body parses.

jadams-tresys commented:
I've reverted my original changes and moved whitespace changes to a separate commit to be easier to review.

As for the actual changes to the schema, it is now using nested sequences to maintain the same behavior for discriminators due to the changes made for DAFFODIL-1971.  Essentially the discriminators being used here were always set to true.  Prior to DAFFODIL-1971 the discriminator expression was only evaluated if the rest of the sequence successfully parsed.  Now the discriminator expression will always be attempted regardless of the reset of the sequence.

stevedlawrence commented on src/main/resources/EDIFACT-Common/IBM_EDI_Format.xsd:
```diff
@@ -1,63 +1,63 @@
-    * IBM hereby grant permission, to those obtaining a copy of these materials, 
-    * to use, copy and modify the materials solely for use with DFDL processors. 
-    * THESE MATERIALS ARE PROVIDED AS IS WITH NO WARRANTIES OF ANY KIND. 
+    * IBM hereby grant permission, to those obtaining a copy of these materials,
```
I'm wondering if there is value in removing the whitespace commit for this repo? For example, maybe we'll want to merge daffodil-dev into main, or diff the files on main to see what changed specifically for Daffodil. It would probably make viewing diffs or merging much easier if we minimized whitespace changes.

jadams-tresys replied:
I was planning on not squashing that commit and leaving it in the commit history so that the changes shouldn't be too bad to reason about.

stevedlawrence replied:
Even if it's not squashed, the files still have 1000+ lines of whitespace changes making them difficult to diff agains files in the main branch. For example, `git diff master --find-renames` will show the diffs between daffodil-dev of those files. If daffodil-dev has a bunch of whitespace changes, that diff would be much more difficult to make sense of.

mbeckerle replied:
I tend to want to avoid the whitespace changes because the Daffodil version of this schema is effectively a "fork" of the original IBM-created EDIFACT schema. Enhancements to this schema coming from the IBM community are likely to be contributed to the main IBM branch. We, or somebody, would then need to map those forward into the Daffodil branch. 

There are two ways that could go. (1) isolate their changes and map forward to the daffodil-dev branch (2) isolate all our changes on the daffodil-dev branch and reapply them to a new copy of the main branch. 

Either way whitespace changes make this more difficult. 

So I suggest roll back the whitespace-only changes also.

stevedlawrence reviewed (approved):
+1, though I wonder if it's worth removing the whitespace changes?

mbeckerle reviewed (approved):
+1
